### ISSUE TITLE
SDK_V0.9 checksum error

### ISSUE THREAD

**Issue by Gimmi7** [fhstp/SoniTalk#1]
I`ve tried the demo app on google play , but the checksum always wrong wahtever the frequency seted. Is there any solution for me? 
thanks a lot

**Comment by aringot** [comment]
Hello, thanks for the feedback!

I would suggest trying with a smaller message (reduce "Maximum message length" to 6 or 8 bytes). A potential problem is movement, so please try to put the phone on a table for sending/receiving.

If it still does not work I would need more details:
What is your operating system, device model, and are you sending/receiving with the same device/configuration ?

Best,
Alexis

**Comment by dusterbloom** [comment]
Hi @aringot 

I've been testing the app on a Nokia 3 ( - Android v9 (patched 05 October) same problem even if I set "Maximum message length" to 08 bytes. 
I am sending with the same configuration. 

I ran into the same problem while emulating the app on Android Studio.

If you need any logs, let me know how we can help.

Thanks

**Comment by nicolas-f** [comment]
I have several smartphones and I have never been able to forward a single message to date. At best, I got a message about a CRC problem. Do you need a special speaker?

**Comment by aringot** [comment]
Hi, we use no special loudspeakers, only standard smartphones. For instance, the Motorola Z works very well for both sending and receiving, whereas the Google Pixel 2 works better for receiving than for sending (its loudspeaker cannot produce strong signals in the higher frequencies). @nicolas-f , may I ask which smartphones you tried already?

It would be interesting to know if your phones are able to send ultrasounds. Could you please send us the spectrogram of a message sent with SoniTalk from your device?

**Comment by nicolas-f** [comment]
I tried with Nexus 6P and Moto E (2nd Gen)

I've tried with Audible and Inaudible configuration.

I will try to record and share some audio files

**Comment by dusterbloom** [comment]
I've tried Nokia 3 (which works just fine with inaudibles coming from either chirp / sonar@x / even quietjs). 

I can send you a wave file if you wish.

**Comment by aringot** [comment]
Thank you, audio files would be great!

**Comment by nicolas-f** [comment]
It works with preset hearable4000.json and volume at 20%

**Comment by aringot** [comment]
@peppilittera @nicolas-f @Gimmi7 , we published a beta version on the play store, could you please confirm if it solves your issue?
You can opt in here: https://play.google.com/apps/testing/at.ac.fhstp.sonitalk.sonitalkdemo
You can also build from the release tagged SDK_v0.9.3: https://github.com/fhstp/SoniTalk/releases/tag/SDK_v0.9.3

**Comment by nicolas-f** [comment]
With beta version It worked 6 times out of 10 attempts. Much more than the release version.

**Comment by aringot** [comment]
Thank you @nicolas-f. I close this issue for now, feel free to reopen or start a new issue if problems appear for specific configurations.

**Comment by rahulshyokand** [comment]
Hi,
I tried to send some text using the beta version. Error: CRC Checksum Incorrect was seen many times as I tried sending and receiving. In fact, my experiment couldn't get data over audio even single time. Also, the speaker had to be very close to the microphone of devices to even get an error message. 
I recorded video while doing the experiment which can be found in the zip file below.
[error-crc-sonitalk.zip](https://github.com/fhstp/SoniTalk/files/4661018/error-crc-sonitalk.zip)
I really appreciate SoniTalk Project. 
Thank you

> @peppilittera @nicolas-f @Gimmi7 , we published a beta version on the play store, could you please confirm if it solves your issue?
> You can opt in here: https://play.google.com/apps/testing/at.ac.fhstp.sonitalk.sonitalkdemo
> You can also build from the release tagged SDK_v0.9.3: https://github.com/fhstp/SoniTalk/releases/tag/SDK_v0.9.3
